src/App.ts

```diff
@@ -314,6 +322,27 @@ export default class App {
     );
   }
 
+  public shortcut<Shortcut extends SlackShortcut = SlackShortcut>(
+    callbackId: string | RegExp,
+    ...listeners: Middleware<SlackShortcutMiddlewareArgs<Shortcut>>[]
+  ): void;
+  public shortcut<Shortcut extends SlackShortcut = SlackShortcut>(
+    constraints: ShortcutConstraints,
+    ...listeners: Middleware<SlackShortcutMiddlewareArgs<Shortcut>>[]
+  ): void;
+  public shortcut<Shortcut extends SlackShortcut = SlackShortcut>(
+    callbackIdOrConstraints: string | RegExp | ShortcutConstraints,
+    ...listeners: Middleware<SlackShortcutMiddlewareArgs<Shortcut>>[]
+  ): void {
+    const constraints: ShortcutConstraints =
+      (typeof callbackIdOrConstraints === 'string' || util.types.isRegExp(callbackIdOrConstraints)) ?
+        { callback_id: callbackIdOrConstraints } : callbackIdOrConstraints;
+
```

Comment:
it seems we're not dong the "fail early if constraints contain invalid keys" here. any reason?